docs/api/tutorials/custom-assertions.md

```diff
@@ -0,0 +1,290 @@
+            # Optional: If you want to provide a custom SQL query for the assertion. This will be rendered as a query in the UI. 
+            # logic: "SELECT * FROM X WHERE Y"
+      }
+  ) {
+      urn
+    }
+}
+```
+
+Note that you can either provide a unique `urn` for the assertion, which will be used to generate the corresponding assertion urn in the following format:
+
+`urn:li:assertion:<your-new-assertion-id>`
+
+or a random urn will be created and returned for you. This id should be stable over time and unique for each assertion.
+
+The upsert API will return the unique identifier (URN) for the the assertion if you were successful:
+
+```json
+{
+  "data": {
+    "upsertExternalAssertion": {
+      "urn": "urn:li:assertion:your-new-assertion-id"
+    }
+  },
+  "extensions": {}
+}
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+To upsert an assertion in Python, simply use the `upsert_external_assertion` method on the DataHub Client object. 
+
+```python
+{{ inline /metadata-ingestion/examples/library/upsert_custom_assertion.py show_path_as_comment }}
+```
+
+</TabItem>
+
+</Tabs>
+
+## Report Results For Custom Assertions
+
+When an assertion is evaluated against a Dataset, or a new result is available, you can report the result to DataHub
+using the following APIs. 
+
+Once reported, these will appear in the evaluation history of the assertion and will be used to determine whether the assertion is
+displayed as passing or failing in the DataHub UI.
+
+<Tabs>
+<TabItem value="graphql" label="GraphQL" default>
+
+To report results for a custom, use the `reportAssertionResult` GraphQL Mutation. This mutation both allows you to
+create and update a given assertion.
+
+```graphql
+mutation reportAssertionResult {
+    reportAssertionResult(
+        urn: "urn:li:assertion:<your-new-assertion-id>"
+        result: {
+            timestampMillis: 1620000000, # Unix timestamp in millis. If not provided, the current time will be used.
+            type: SUCCESS,  # or FAILURE or ERROR or INIT
+            properties: [
+                {
+                    key: "my_custom_key",
+                    value: "my_custom_value"
+                }
+            ],
+            externalUrl: "https://my-great-expectations.com/results/1234", # Optional: URL to the results in the external tool
+            # Optional: If the type is ERROR, you can provide additional context. See full list of error types below. 
+            # error: {
+            #    type: UNKNOWN_ERROR,
+            #    message: "The assertion failed due to an unknown error"
+            # }
+      }
+  )
+}
+```
+
+The `type` field is used to communicate the latest health status of the assertion.
+
+The `properties` field is used to provide additional key-value pair context that will be displayed alongside the result
+in DataHub's UI. 
+
```

Comment:
@jjoyce0510  Does `properties` represent input passed in when running the assertion in platform  (`runtimeContext` aka runtime variables) or native results dictionary (`nativeResults`) ?

Reply:
yes!